lib/index.js

```diff
@@ -169,10 +169,18 @@ Assertion.prototype.maps = function(fixture, settings, options){
   actual = JSON.parse(JSON.stringify(actual));
 
   if (options.ignored) {
-    options.ignored.map(function(name) {
-      delete actual[name]
-      assert(!expected[name], "ignored expected value:" + name);
-    });
+    // Recursively strip all keys given in options.ignored from actual, and from
+    // all objects contained in or nested under actual.
+    var stripIgnoredFromResult = function(obj) {
+      for(prop in obj) {
+        if(options.ignored.indexOf(prop) != -1) {
+          delete obj[prop];
+        } else if (typeof obj[prop] === 'object') { // recurse on contained objs
```

Comment:
You might want to check if it's an array too. Arrays are also objects and `typeof [] === 'object'` is `true`.

```js
else if (typeof obj[prop] === 'object' && !(obj[prop] instanceof Array)) {
  // ...
}
```